### PR TITLE
simplify entity restriction and enhance performance

### PR DESCRIPTION
**Pull request by btry**
It seems there is a bad design here, and it heavily impacts performance when GLPI has a lot of entities

This PR should fix it

### CODE REVIEW

File: inc/form.class.php [hunk 1]
```diff
@@ -336,12 +336,7 @@ public function showForm($ID, $options=array()) {
 
       // Si le formulaire est récursif, on authorise les validateurs des sous-entités
       // Sinon uniquement les validateurs de l'entité du formulaire
-      if ($this->isRecursive()) {
-         $entites = getSonsOf('glpi_entities', $this->getEntityID());
-      } else {
-         $entites = $this->getEntityID();
-      }
-      $subentities = getEntitiesRestrictRequest("", 'pu', "", $entites, true, true);
+      $subentities = getEntitiesRestrictRequest("", 'pu', "", $this->getEntityID(), $this->isRecursive() == 1, true);
```
Comment (orthagh) [inline inc/form.class.php]: ok for replacement

But why use ``== 1``, seems superflu

Comment (wawax) [reply inc/form.class.php]: seems ok to me also. Same comment as @orthagh

File: inc/form.class.php [hunk 2]
```diff
@@ -336,12 +336,7 @@ public function showForm($ID, $options=array()) {
 
       // Si le formulaire est récursif, on authorise les validateurs des sous-entités
       // Sinon uniquement les validateurs de l'entité du formulaire
```
Comment (wawax) [inline inc/form.class.php]: The comment should be in english

File: inc/form.class.php (hunk 1 above)
Comment (btry) [reply inc/form.class.php]: == 1 has two purposes : cast into a boolean, and also ensure true is returned only when the column is_recursive contains 1.
I think ```$this->isRecursive() != 0``` is even better (this is an habit I adopted in Flyve MDM plugin for example). I avoid use of empty() to keep things simple.

Comment (orthagh) [reply inc/form.class.php]: you do exactly the same thing as php core.
If you want to explicitly cast (i still think it's useless here), do it but with a real cast syntax.

**Comment by btry** [comment]
Planning to add unit tests with atoum, for 2.7.0

**Comment by btry** [comment]
Outdated by code refactors

**Pull request closed without merging**